block/provider/list_snapshots.go

```diff
@@ -18,13 +18,76 @@
 package provider
 
 import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/IBM/ibmcloud-volume-interface/lib/metrics"
 	"github.com/IBM/ibmcloud-volume-interface/lib/provider"
+	userError "github.com/IBM/ibmcloud-volume-vpc/common/messages"
+	"github.com/IBM/ibmcloud-volume-vpc/common/vpcclient/models"
+	"go.uber.org/zap"
 )
 
+const startSnapshoIDNotFoundMsg = "start parameter is not valid"
+
 // ListSnapshots list all snapshots
-func (vpcs *VPCSession) ListSnapshots() ([]*provider.Snapshot, error) {
+func (vpcs *VPCSession) ListSnapshots(limit int, start string, filters map[string]string) (*provider.SnapshotList, error) {
 	vpcs.Logger.Info("Entry ListeSnapshots")
 	defer vpcs.Logger.Info("Exit ListSnapshots")
+	defer metrics.UpdateDurationFromStart(vpcs.Logger, "ListSnapshots", time.Now())
+
+	if limit < 0 {
+		return nil, userError.GetUserError("InvalidListSnapshotLimit", nil, limit)
+	}
+
+	if limit > maxLimit {
+		vpcs.Logger.Warn(fmt.Sprintf("listSnapshots requested max entries of %v, supports values <= %v so defaulting value back to %v", limit, maxLimit, maxLimit))
+		limit = maxLimit
+	}
+
+	filter := &models.LisSnapshotFilters{
+		ResourceGroupID: filters["resource_group.id"],
+		Name:            filters["name"],
+		SourceVolumeID:  filters["source_volume.id"],
+	}
+
+	vpcs.Logger.Info("Getting snapshot list from VPC provider...", zap.Reflect("start", start), zap.Reflect("filters", filters))
+
+	var snapshots *models.SnapshotList
```

Comment:
as this will be in memory, so we need to see how many snapshot can be fetch, this would be a future enhancement

Reply:
https://github.ibm.com/alchemy-containers/armada-storage/issues/3801